src/nisarqa/raster_classes.py

```diff
@@ -499,6 +505,30 @@ def x_axis_limits(self) -> tuple[float, float]:
     def x_axis_label(self) -> str:
         return f"X Coordinate, EPSG:{self.epsg} (km)"
 
+    @property
+    def x_start(self) -> float:
+        return float(self.x_coordinates[0])
+
+    @property
+    def x_stop(self) -> float:
+        # X in meters (units are specified as meters in the product spec)
+        # For NISAR, geocoded grids are referenced by the upper-left corner
+        # of the pixel to match GDAL conventions. So add the distance of
+        # the pixel's side to far right side to get the actual stop value.
+        return float(self.x_coordinates[-1] + self.x_axis_spacing)
```

Comment:
Oops, we were wrong about this. Let's fix this in a future PR.

The subtlety here is that `isce3.product.GeoGridParameters` is referenced to the upper-left corner of the first pixel, *but* the `xCoordinates` and `yCoordinates` datasets that we store in NISAR products don't follow this convention. Instead, they represent the center of each pixel in the grid.

(See here for reference: https://github.com/gmgunter/isce3/blob/e4e4e568487d4e6c37f63c323eace51e723e4397/python/packages/nisar/static/product.py#L382-L395)